### PR TITLE
[#52] update CI pipeline

### PR DESCRIPTION
**Pull request by rvem**
## Description
Add hlint and weeder steps to the CI pipeline.

TODO: figure out whether we want to use new weeder as well.

## Related issue(s)

Resolves partially #52 

#### Related changes (conditional)

- Tests
  - [x] If I added new functionality, I added tests covering it.
  - [x] If I fixed a bug, I added a regression test to prevent the bug from
        silently reappearing again.

- Documentation
  - [x] I checked whether I should update the docs and did so if necessary:
    - [README](../tree/master/README.md)
    - Haddock
  - [x] I updated [changelog](../tree/master/ChangeLog.md) unless I am sure my changes are
        not essential.

#### Stylistic guide (mandatory)

- [x] My commits comply with [the policy used in Serokell](https://www.notion.so/serokell/Where-and-how-to-commit-your-work-58f8973a4b3142c8abbd2e6fd5b3a08e).
- [x] My code complies with the [style guide](../tree/master/docs/code-style.md).

### CODE REVIEW

File: default.nix [hunk 1]
```diff
@@ -4,16 +4,29 @@
...
+        packages = pkgs.lib.genAttrs local-packages-names (packageName: {
+            package.ghcOptions = with pkgs.lib;
+              concatStringsSep " " ([
+                "-ddump-to-file" "-ddump-hi"
+              ]);
+            postInstall = weeder-hacks.collect-dump-hi-files;
+        });
+        # packages.stablecoin.components.tests.stablecoin-test = {
+        #   postInstall = "mkdir -p $out/test/resources; cp -Lr ${tezos-contract} $out/test/resources/stablecoin.tz";
+        # };
```
Comment (gromakovsky) [inline default.nix:29]: Is it needed?

**Comment by gromakovsky** [comment]
> Resolves partially #52

Because of new weeder?

> TODO: figure out whether we want to use new weeder as well.

Did you try doing that? I would try and if it appears to be straightforward – let's do it, otherwise let's forget about it for now. I suppose one day we will have to switch to new `weeder` (or just stop using the old one because it's not maintained, but not start using the new one), so our first step should be to check whether it's easy to setup.

**Comment by rvem** [comment]
> Because of new weeder?
Yes
> Did you try doing that?
Currently, I'm trying to build it and test locally

**Comment by rvem** [comment]
```
$ ./weeder
src/Indigo/Contracts/Safelist.hs:28:1: error: mkStorage is unused

      26 weeder: <stdout>: commitBuffer: invalid argument (invalid character)
```
Sad that it doesn't produce fancy output like in https://github.com/ocharles/weeder#calling-weeder

**Comment by rvem** [comment]
okay, looks like new weeder requires the presence of `.hie` files, thus we will need some additional workarounds to make it work with haskell.nix, let's do it in a separate PR

**Comment by gromakovsky** [comment]
> Resolves partially #52

@rvem why partially? Because of new `weeder`? I think it's not a big problem, using only one `weeder` should be fine. We can get back to this question one the old `weeder` dies.

**Comment by rvem** [comment]
Okay, let's return to this once the old weeder is deprecated

**Pull request merged**